docs/introduction.asciidoc

```diff
@@ -5,29 +5,31 @@ https://www.elastic.co/guide/en/apm/agent/ruby/current/introduction.html[elastic
 endif::[]
 
 [[introduction]]
-
 == Introduction
 
-Welcome to the APM Ruby Agent documentation.
-
-The Elastic APM Ruby Agent sends performance metrics and error logs to an
-Elastic APM Server.
+The Elastic APM Ruby Agent sends performance metrics and error logs to the APM Server.
 It has built-in support for <<getting-started-rails,Ruby on Rails>> and other
 <<getting-started-rack,Rack-compatible>> applications.
+It also offers an API which allows you to instrument any application.
+
+[float]
+[[how-it-works]]
+=== How does the Agent work?
+
+The agent auto-instruments <<supported-technologies,supported technologies>> and records interesting events,
+like HTTP requests and database queries. To do this, it {NEED CONTENT HERE}.
```

Comment:
```suggestion
like HTTP requests and database queries. To do this, it uses relevant public APIs where such are provided by the libraries or, if they aren't, carefully wraps the necessary internal methods.
```

Reply:
maybe break it up into two sentences? `uses relevant public APIs when they are provided by the libraries. Otherwise, it carefully wraps the necessary internal methods.`

Reply:
Sure – try a one of those code suggestion comments :)